docs/manual/resource-leak-checker.tex

```diff
@@ -0,0 +1,369 @@
+
+\item[\refqualclasswithparams{checker/mustcall/qual}{InheritableMustCall}{String[] value}]
+on any classes defined in your program that have must-call obligations. See Section~\ref{must-call-on-class}.
+
+\item[\refqualclass{checker/calledmethods/qual}{EnsuresCalledMethods}] on a method in
+your code that fulfills a must-call obligation of one of its parameters or of a field.
+See Section~\ref{called-methods-ensurescalledmethods}.
+
+\end{description}
+
+The Resource Leak Checker supports annotations that express
+aliasing patterns related to resource leaks:
+
+\begin{description}
+
+\item[\refqualclass{checker/mustcall/qual}{Owning}]
+\item[\refqualclass{checker/mustcall/qual}{NotOwning}]
+  expresses ownership.  When two aliases exist to the same Java object,
+  \<@Owning> and \<@NotOwning> indicate which of the two is responsible for
+  fulfilling must-call obligations.  For details, see
+  Section~\ref{resource-leak-ownership}.
+
+\item[\refqualclass{checker/mustcall/qual}{MustCallAlias}]
+  represents a ``resource-aliasing'' relationship.  Resource aliases are
+  distinct Java objects that control the same resource(s):
+  fulfilling the must-call obligations of one object also
+  fulfills the obligations of the other object.  For details,
+  see Section~\ref{resource-leak-resource-alias}.
+
+\end{description}
+
+The Resource Leak Checker also supports an annotation to permit re-assigning
+fields or re-opening resources:
+
+\begin{description}
+
+\item[\refqualclasswithparams{checker/mustcall/qual}{CreatesMustCallFor}{String value}]
+  is a declaration annotation that indicates that after a call to a method
+  with this annotation none of the must-call obligations of the in-scope, owning expression
+  listed in \<value> have been met.
+  In other words, the annotated method ``resets'' the must-call obligations of the expression.
+  Multiple \<@CreatesMustCallFor>
+  annotations can be written on the same method.  For more details
+  on how to use this annotation to permit re-assignment of owning
+  fields or the re-opening of resources, see Section~\ref{resource-leak-createsmustcallfor}.
+
+\end{description}
+
+\sectionAndLabel{Example of how safe resource usage is verified}{resource-leak-example}
+
+Consider the following example of a safe use of a \<Socket>:
+\begin{verbatim}
+Socket s = null;
+try {
+  s = new Socket(myHost, myPort);
+} catch (Exception e) {
+  // do nothing
+} finally {
+  if (s != null) {
+    s.close();
+  }
+}
+\end{verbatim}
+
+On the first line, \<s> is assigned to \<null>. At this point, it has a type from
+each of the Must Call Checker and the Called Methods Checker: \<@MustCall(\{\})> and
+\<@CalledMethodsBottom>, respectively (because \<null> has no must-call obligations
+and methods cannot be called on it).
+
+Inside the \<try> block, \<s> is assigned to a new \<Socket> object. New \<Socket> objects
+do have a must-call obligation and methods can be called on them, so \<s>'s types at the end of the \<try>
```

Comment:
I'm not clear on whether "end of the `try`" means the try body or the try statement.